### DISCUSSION TITLE
Nozzle Prime After Filament Change

### DISCUSSION THREAD

**Discussion by reyesv351** [mriscoc/Ender3V2S1#981] [Ideas]
Not sure if it would be possible to add the feature into the menu to allow the user to change the amount the nozzle primes after a filament change during a print. 

For example the user starts the print, slicer pauses, unloads during the advance nozzle park, filament feed, then resume print but the custom amount the user sets is primed in the nozzle when resuming the print where it leaves off. 

I set the extra prime on resume when building my E5 FW but would like to see it added to your professional FW in a submenu _if_ possible.

Thanks for all your hard work and contribution to the community by the way!

**Comment by gothcha** [comment]
Hi, under Advanced Settings - Filament Settings if you scroll down there is a Load mm & Unload mm setting which I believe should achieve what you are looking for?

**Reply by reyesv351** [reply]
Hello and thank you but, that only sets the amount for the printer to load when using the LCD to load or unload filament when not printing. During mid print there's a setting within the firmware to set and additional nozzle prime when resuming. ADVANCED_PAUSE_RESUME_PRIME on my direct drive Ender 5 P is where I set an additional 1.5 mm prime. I'm not sure if Mr Castillo can add the feature to the LCD Menu or not but it would certainly help those that don't know how to build the FW.

**Reply by gothcha** [reply]
OK understood, I have not yet tried the filament change at layer, but I will need to use it soon. I had assumed it would use the same values as the manual load/unload, I am still quite new to this firmware.

**Reply by reyesv351** [reply]
No problem, you can push a little more filament into the nozzle by squeezing the extruder tension and pushing in the filament a bit if you catch it in time but if you're not careful you'll push a little too much and it will smoosh into the print, which sometimes can still be cleaned up with a hobby knife.

**Reply by reyesv351** [reply]
Filament changes in the slicer can be pretty awesome if you can get things going. Not trying to advertise here or anything but there's some software out there (it's seriously awesome what you can do with it), it's called Hue Forge that allows you to essentially paint an image with filament but you need to have the ability to pause and change filament which his firmware does. Just unfortunately for those who don't have the knowledge yet to build firmware would have some difficulty with the filament change and not having the nozzle primed unless they really get the hang of it.

**Comment by mriscoc** [comment]
>ADVANCED_PAUSE_RESUME_PRIME on my direct drive Ender 5 P is where I set an additional 1.5 mm prime.

For 1.5mm you can increase a little the filament load length in the filament settings menu. That length value is used both by the menu actions and by filament change G-code M600.

ADVANCED_PAUSE_RESUME_PRIME  is used for the repetitive purge feature after the load.

**Reply by reyesv351** [reply]
Hello Mr. Risco-Castillo thanks for your reply. I put in a 1.5 mm load length in the filament menu and even a 10mm load but it did not make a difference when the print resumed. The print head continued on without filament and eventually moved the extruder along enough that it began printing with filament.

**Reply by mriscoc** [reply]
The filament load length is approximately the distance between the input of the extruder and the tip of the nozzle. In a bowden system, for example, it is the length of the PTFE tube plus about 100mm.

**Reply by reyesv351** [reply]
![capture](https://github.com/mriscoc/Ender3V2S1/assets/138394014/57656349-d086-4712-b1d4-b0d6cddbe4a6)

 this is where I mentioned putting in the extra length.
 I instead of leaving the filament at the extruder push the length all the way through to the nozzle and then select the feed option given to mitigate the bowden tube. The ADVANCED_PAUSE_RESUME_PRIME is where the extra length is pushed after the entire Advanced Pause is completed and only takes place where the nozzle is supposed to resume printing. I'm not sure if it would be possible for you to add the option to change this value into a sub menu, but it would be extremely helpful. Thank you again for hearing me out.

**Reply by reyesv351** [reply]
If it's not possible to add this to a sub menu I completely understand. I just thought it would be a great addition to the LCD sub menus as an option.
